src/net_processing.cpp

```diff
@@ -5669,7 +5721,42 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         if (tx_relay->m_bloom_filter && !tx_relay->m_bloom_filter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         // Send
                         State(pto->GetId())->m_recently_announced_invs.insert(hash);
-                        vInv.push_back(inv);
+
+                        bool fanout = true;
+                        if (reconciles_txs) {
+                            LOCK(m_mempool.cs);
+                            auto txiter = m_mempool.GetIter(txinfo.tx->GetHash());
+                            assert(txiter);
+                            if ((*txiter)->GetCountWithDescendants() > 1) {
+                                // If a transaction has in-mempool children, always fanout it.
+                                // Until package relay is implemented, this is needed to avoid
+                                // breaking parent+child relay expectations in some cases.
+                                //
+                                // Potentially reconciling parent+child would mean that for every
+                                // child we need to to check if any of the parents is currently
+                                // reconciled so that the child isn't fanouted ahead. But then
+                                // it gets tricky when reconciliation sets are full: a) the child
+                                // can't just be added; b) removing parents from reconciliation
+                                // sets for this one child is not good either.
+                                fanout = true;
+                            } else {
+                                size_t inbounds_fanouted = 0, outbounds_fanouted = 0;
+                                m_connman.ForEachNode([&inbounds_fanouted, &outbounds_fanouted, this](CNode* pnode) {
+                                    inbounds_fanouted += pnode->IsInboundConn() && pnode->m_relays_txs;
+                                    outbounds_fanouted += pnode->IsFullOutboundConn() && pnode->m_relays_txs && !m_txreconciliation->IsPeerRegistered(pnode->GetId());
+                                });
```

Comment:
```suggestion
                                {
                                    LOCK(m_peer_mutex);
                                    for (auto [id, peer] : m_peer_map) {
                                        const auto state{State(id)};
                                        if (!state) continue;

                                        if (auto tx_relay = peer->GetTxRelay()) {
                                            LOCK(tx_relay->m_bloom_filter_mutex);
                                            inbounds_fanouted += state->m_is_inbound && tx_relay->m_relay_txs;
                                            outbounds_fanouted += !state->m_is_inbound && tx_relay->m_relay_txs && !m_txreconciliation->IsPeerRegistered(id);
                                        }
                                    }
                                }

```

See https://github.com/bitcoin/bitcoin/pull/27270